mysql/README.md

```diff
@@ -4,40 +4,146 @@
+
+<!-- xxx tabs xxx -->
+<!-- xxx tab "MySQL ≤ 5.7" xxx -->
+
+For managed databases, replace `localhost` in every occurrence of `'datadog'@'localhost'` with the host where your agent will be running, or put `%` to allow the agent to connect from any host. For more information, see the [MySQL documentation][5].
+
+```SQL
+CREATE USER 'datadog'@'localhost' IDENTIFIED BY '<UNIQUEPASSWORD>';
+GRANT REPLICATION CLIENT ON *.* TO 'datadog'@'localhost' WITH MAX_USER_CONNECTIONS 5;
+GRANT PROCESS ON *.* TO 'datadog'@'localhost';
+GRANT SELECT ON performance_schema.* TO 'datadog'@'localhost';
+GRANT CREATE TEMPORARY TABLES ON `datadog`.* TO 'datadog'@'localhost';
 ```
 
-For mySQL 8.0+ create the `datadog` user with the native password hashing method:
+<!-- xxz tab xxx -->
+<!-- xxx tab "MySQL ≥ 8.0" xxx -->
+
+For managed databases, replace `localhost` in every occurrence of `'datadog'@'localhost'` with the host where your agent will be running, or put `%` to allow the agent to connect from any host. For more information, see the [MySQL documentation][5]. 
+
+```SQL
+CREATE USER 'datadog'@'localhost' IDENTIFIED WITH mysql_native_password by '<UNIQUEPASSWORD>';
+ALTER USER 'datadog'@'localhost' WITH MAX_USER_CONNECTIONS 5;
+GRANT REPLICATION CLIENT ON *.* TO 'datadog'@'localhost'
+GRANT PROCESS ON *.* TO 'datadog'@'localhost';
+GRANT SELECT ON performance_schema.* TO 'datadog'@'localhost';
 
-```shell
-mysql> CREATE USER 'datadog'@'localhost' IDENTIFIED WITH mysql_native_password by '<UNIQUEPASSWORD>';
-Query OK, 0 rows affected (0.00 sec)
 ```
 
-**Note**: `@'localhost'` is only for local connections - use the hostname/IP of your Agent for remote connections. For more information, see the [MySQL documentation][5].
+<!-- xxz tab xxx -->
+<!-- xxz tabs xxx -->
+
+Create the functions to enable collection of statements & execution plans. This is required only for [Deep Database Monitoring](#deep-database-monitoring).
+
+```SQL
+CREATE SCHEMA datadog;
+GRANT CREATE TEMPORARY TABLES ON `datadog`.* TO 'datadog'@'localhost';
+
+DELIMITER $$
+CREATE PROCEDURE datadog.explain_statement(IN query TEXT)
+    SQL SECURITY DEFINER
+BEGIN
+    SET @explain := CONCAT('EXPLAIN FORMAT=json ', query);
+    PREPARE stmt FROM @explain;
+    EXECUTE stmt;
+    DEALLOCATE PREPARE stmt;
+END $$
+
+-- (Optional) Gives the agent the ability to enable performance schema events_statements consumers on databases where they can't be enabled permanently in the configuration (like RDS Aurora)
+CREATE PROCEDURE datadog.enable_events_statements_consumers()
+    SQL SECURITY DEFINER
+BEGIN
+    UPDATE performance_schema.setup_consumers SET enabled='YES' WHERE name LIKE 'events_statements_%';
+END $$
+
+DELIMITER ;
+
+GRANT EXECUTE ON PROCEDURE datadog.explain_statement TO 'datadog'@'localhost';
+GRANT EXECUTE ON PROCEDURE datadog.enable_events_statements_consumers TO 'datadog'@'localhost';
+```
```

Comment:
I believe this one should be a separate SQL block with paragraph text around it.

Reply:
Agreed, will update